### PR TITLE
Slasher improvements

### PR DESCRIPTION
**Pull request by shayzluf**
**What type of PR is this?**

 Bug fix

**What does this PR do? Why is it needed?**
this pr fixes:
- reconnect open files issue
- add log for rpc server calls
- sets spans-cache-size to 1500 by default
- remove traces to improve performance

### CODE REVIEW

File: slasher/rpc/server.go [hunk 1]
```diff
@@ -37,6 +39,10 @@ func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Indexed
 	ctx, span := trace.StartSpan(ctx, "detection.IsSlashableAttestation")
 	defer span.End()
 
+	log.WithFields(logrus.Fields{
+		"slot":    req.Data.Slot,
+		"indices": req.AttestingIndices,
+	}).Debug("Server attestation detection")
```
Comment (0xKiwi) [inline slasher/rpc/server.go:45]: Can you change this log to be a bit more descriptive? Maybe like "Received attestation via RPC"?

File: slasher/rpc/server.go [hunk 2]
```diff
@@ -116,6 +122,10 @@ func (ss *Server) IsSlashableBlock(ctx context.Context, req *ethpb.SignedBeaconB
 	ctx, span := trace.StartSpan(ctx, "detection.IsSlashableBlock")
 	defer span.End()
 
+	log.WithFields(logrus.Fields{
+		"slot":           req.Header.Slot,
+		"proposer_index": req.Header.ProposerIndex,
+	}).Info("Server block for detection")
```
Comment (0xKiwi) [inline slasher/rpc/server.go:128]: Same here

File: slasher/rpc/server.go [hunk 3]
```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 	"sync"
 
+	"github.com/sirupsen/logrus"
+
```
Comment (0xKiwi) [inline slasher/rpc/server.go:8]:
```suggestion
```

File: slasher/beaconclient/receivers.go [hunk 4]
```diff
@@ -173,61 +184,31 @@ func (bs *Service) collectReceivedAttestations(ctx context.Context) {
...
 	ticker := time.NewTicker(reconnectPeriod)
 	for {
 		select {
 		case <-ticker.C:
-			log.Info("Context closed, attempting to restart attestation stream")
-			conn, err := grpc.DialContext(bs.ctx, bs.provider, bs.beaconDialOptions...)
-			if err != nil {
-				log.Debug("Failed to dial beacon node")
+			if bs.conn.GetState() == connectivity.TransientFailure || bs.conn.GetState() == connectivity.Idle {
+				log.Debugf("connection status %v", bs.conn.GetState())
```
Comment (0xKiwi) [inline slasher/beaconclient/receivers.go:193]:
```suggestion
				log.Debugf("Connection status %v", bs.conn.GetState())
```

File: slasher/rpc/server.go (hunk 1 above)
Comment (shayzluf) [reply slasher/rpc/server.go:45]: changed

File: slasher/rpc/server.go (hunk 2 above)
Comment (shayzluf) [reply slasher/rpc/server.go:128]: changed

File: slasher/rpc/server.go (hunk 3 above)
Comment (shayzluf) [reply slasher/rpc/server.go:8]: thanks

**Pull request merged**